File: Python_Collections_benchmark/ListTest.py
from IBenchmark import IBenchmark
from collections import UserList


class ListTest(IBenchmark):
    ls = UserList()
    num_of_operations = 100000

    def __init__(self, num_of_operations):
        self.num_of_operations = num_of_operations

    def add_test(self):
        for i in range(0, self.num_of_operations):
            to_add = len(self.ls)
            self.ls.append(to_add)

    def find_test(self):
        to_find = len(self.ls) - 1
        for i in range(0, self.num_of_operations):
            result = to_find in self.ls

    def remove_test(self):
        for i in range(0, self.num_of_operations):
            to_remove = self.ls[0]
            self.ls.remove(to_remove)

File: CS_Collections_benchmark/ListTest.cs
using System;
using System.Collections.Generic;
using System.Text;

namespace CS_Collections_benchmark
{
    class ListTest : IBenchmark
    {
        List<int> list;
        int numOfOperations;

        public ListTest()
        {
            list = new List<int>();
            numOfOperations = 100;
        }

        public ListTest(int numOfOperations)
        {
            list = new List<int>();
            this.numOfOperations = numOfOperations;
        }

        public void AddTest()
        {
            for (int i = 0; i < numOfOperations; i++)
            {
                list.Add(i);
            }
        }

        public void FindTest()
        {
            int toFind = list[list.Count - 1];
            for (int i = 0; i < numOfOperations; i++)
            {
                int foundElement = list.Find(x => x == toFind);
            }
        }

        public void RemoveTest()
        {
            for (int i = 0; i < numOfOperations; i++)
            {
                int toRemove = list[0];
                list.Remove(toRemove);
            }
        }
    }
}
